axelrod/strategies/titfortat.py

```diff
@@ -429,3 +429,79 @@ def strategy(self, opponent):
 
         # Otherwise cooperate
         return C
+        
+class AdaptiveTitForTat(Player):
+    """ATFT - Adaptive Tit for Tat (Basic Model)
+    
+    Algorithm
+    ----------
+    if (opponent played C in the last cycle) then
+    world = world + r*(1-world)
+    else
+    world = world + r*(0-world)
+    If (world >= 0.5) play C, else play D
+    
+    Attributes
+    ----------
+    world : float [0.0, 1.0], set to 0.5
+        continuous variable representing the world's image
+        1.0 - total cooperation
+        0.0 - total defection
+        other values - something in between of the above
+        updated every round, starting value shouldn't matter as long as
+        it's >= 0.5
+        
+    Parameters
+    ----------
+    rate : float [0.0, 1.0], default=0.5
+        adaptation rate - r in Algorithm above
+        smaller value means more gradual and robust
+        to perturbations behaviour
+        
+    References
+    ----------
+        http://users.softlab.ntua.gr/~brensham/Publications/SAB2000.pdf
+    """
+    
+    name = 'Adaptive Tit For Tat'
+    classifier = {
+        'memory_depth': float('inf'),
+        'stochastic': False,
+        'makes_use_of': set(),
+        'long_run_time': False,
+        'inspects_source': False,
+        'manipulates_source': False,
+        'manipulates_state': False
+    }
+    world = 0.5
+
+    @init_args
+    def __init__(self, rate=0.5):
+    
+        super().__init__()
```

Comment:
This line is Python 3 only. We're currently also supporting Python 2.7, so we need to use `Player.__init__(self)` instead.